### PR TITLE
Docker setup

### LINKED ISSUE #623: Containerize Tests
- [x] Create a Series of Images as a basline (LLVM inside)
- [x] Create Dockerfiles that build dawn/GTClang based on those
- [x] Investigate how to integrate docker in the testing of jenkins
- [ ] Create a Dockerfile that build gt4py based on the dawn one

### PR DESCRIPTION
**Pull request by jdahm**
## Technical Description

Use Sarus for containerized tests of Dawn.

To do before merging:
- [x] Add Atlas and ecKit
- [x] Create image update script (`scripts/update-images`)
- [x] Check if `sarus pull $image` is the correct usage -- https://github.com/eth-cscs/sarus/issues/6
- [ ] Move `scripts/jenkins/srun_dawn_PR.sh` to `scripts/daint-dawn-PR`
- [ ] Remove ssh:// workaround in git clone URL for clang-gridtools in Jenkins -- this is no longer needed

### Resolves / Enhances

Resolves #623.

### Testing

http://localhost:8080/view/PASCHA/job/dawn_gtclang_clang-gridtools_PR/490/

### CODE REVIEW

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
launch jenkins

File: Dockerfile [hunk 1]
```diff
@@ -1,54 +1,18 @@
...
-    -DGT_INSTALL_EXAMPLES=OFF \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_INSTALL_PREFIX=/usr/local \
-    -GNinja
-RUN cmake --build /usr/src/gridtools-1.0.4/build -j $(nproc) --target install
-RUN rm -rf /usr/src/gridtools-1.0.4/build
-
-FROM dawn-gcc9-env AS dawn-gcc9
+# IMAGE needs be be set to one of the docker/dawn-env.dockerfile images
+ARG IMAGE=gtclang/dawn-env-cuda10.1-ubuntu18.04
```
Comment (twicki) [inline Dockerfile]: if we now have a docker folder, should we put this file also in there (and adapt the path on dockerhub?)

File: .gitignore [hunk 2]
```diff
@@ -24,3 +24,6 @@ _local_/*
 
 # Ignore python editable installs
 *.egg-info/
+
+# Virtual envs
+.venv
```
Comment (twicki) [inline .gitignore]: rm

File: scripts/build-and-test [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    shift
+done
+
+# Dawn
+cmake -S $dawn_source_dir -B $dawn_build_dir \
+    -DBUILD_TESTING=ON \
+    -DCMAKE_INSTALL_PREFIX=$dawn_install_dir \
+    ${build_args[@]}
+
+time cmake --build $dawn_build_dir --config $build_type --parallel $build_jobs --target install
```
Comment (twicki) [inline scripts/build-and-test]: should the timings stay?

**Comment by jdahm** [comment]
launch jenkins

File: scripts/build-and-test (hunk 3 above)
Comment (jdahm) [reply scripts/build-and-test:62]: Probably not. I could make it an option.

File: Dockerfile (hunk 1 above)
Comment (jdahm) [reply Dockerfile:2]: I'm not sure... Sometimes it's nice to have a top-level `Dockerfile` that people can immediately build. Something like a `Makefile` used to be.

Comment (jdahm) [reply Dockerfile:2]: But I'm open to either way! Let me know.

File: scripts/build-and-test (hunk 3 above)
Comment (jdahm) [reply scripts/build-and-test]: Made it an option `-t` or `--timing`.

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
Everything passes so I'm going to go ahead and merge this now.

**Pull request merged**